addons/html_builder/static/src/builder/plugins/builder_overlay/builder_overlay_plugin.js

```diff
@@ -1,34 +1,247 @@
+            selector: ".row > div",
+            exclude: ".s_col_no_resize.row > div, .s_col_no_resize",
+        };
+        this.sizingGrid = {
+            selector: "div.o_grid_item",
+        };
     }
 
-    destroy() {
-        this.removeCurrentOverlay?.();
+    // displayOverlayOptions(el) {
+    //     // TODO when options will be more clear:
+    //     // - moving
+    //     // - timeline
+    //     // (maybe other where `displayOverlayOptions: true`)
+    //     return el.matches([this.sizingY.selector, this.sizingX.selector, this.sizingGrid.selector].join(","));
+    // }
+
+    // TODO improve with different sizing.
+    displayHandles(el) {
+        return el.matches(
+            [this.sizingY.selector, this.sizingX.selector, this.sizingGrid.selector].join(",")
+        );
     }
 
     openBuilderOverlay(toolboxes) {
-        const toolbox = toolboxes[0];
-        this.removeCurrentOverlay?.();
-        if (!toolbox) {
+        this.removeBuilderOverlay();
+        if (!toolboxes.length) {
+            return;
+        }
+        this.toolboxes = toolboxes;
+
+        // Create the overlays.
+        toolboxes.forEach((toolbox) => {
+            const overlayElement = renderToElement("html_builder.BuilderOverlay");
+            const overlayTarget = toolbox.element;
+            this.overlayContainer.append(overlayElement);
+            const displayHandles = this.displayHandles(overlayTarget);
+
+            this.overlays.push(new BuilderOverlay(overlayElement, overlayTarget, displayHandles));
+        });
+
+        // Activate the last overlay.
+        const innermostOverlay = this.overlays.at(-1);
+        innermostOverlay.toggleOverlay(true);
+
+        // Also activate the closest overlay that should have handles.
+        if (!innermostOverlay.displayHandles) {
+            for (let i = this.overlays.length - 2; i >= 0; i--) {
+                const parentOverlay = this.overlays[i];
+                if (parentOverlay.displayHandles) {
+                    parentOverlay.toggleOverlay(true);
+                    break;
+                }
+            }
+        }
+
+        // TODO check if resizeObserver still needed.
+        // this.resizeObserver = new ResizeObserver(this.update.bind(this));
+        // this.resizeObserver.observe(this.overlayTarget);
+    }
+
+    removeBuilderOverlay() {
+        this.overlays.forEach((overlay) => overlay.overlayElement.remove());
+        this.overlays = [];
+        this.toolboxes = [];
+        // this.resizeObserver?.disconnect();
+    }
+
+    _update() {
+        this.overlays.forEach((overlay) => {
+            overlay.refreshPosition();
+            overlay.refreshHandles();
+        });
+    }
+
+    refreshPosition() {
+        this.overlays.forEach((overlay) => {
+            overlay.refreshPosition();
+        });
+    }
+
+    refreshHandles() {
+        this.overlays.forEach((overlay) => {
+            overlay.refreshHandles();
+        });
+    }
```

Comment:
do we still need need with _update

Reply:
I prefer keeping everything separated, because we do not always need to update both at the same time, e.g.:
- scrolling only needs to refresh the position, as the handles are not supposed to change
- using an option may change both the size and the "status" of an element, so we need to refresh both the handles and the position

However, I agree that `refreshHandles` alone is not needed yet, but let's keep it for now to be consistent, we can remove it later if really no case needs it in the end :slightly_smiling_face: